packages/odyssey-react-mui/src/labs/FileUploadPreview.tsx

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, useCallback } from "react";
+import styled from "@emotion/styled";
+import { IconButton } from "@mui/material";
+import { useTranslation } from "react-i18next";
+
+import { FileUploadProps } from "./FileUpload";
+import {
+  useOdysseyDesignTokens,
+  DesignTokens,
+} from "../OdysseyDesignTokensContext";
+import { DeleteIcon } from "../icons.generated";
+import { MuiPropsContext, MuiPropsContextType } from "../MuiPropsContext";
+import { Tooltip } from "../Tooltip";
+
+const PreviewContainer = styled.div<{
+  isDisabled: FileUploadProps["isDisabled"];
+  odysseyDesignTokens: DesignTokens;
+}>(
+  ({ odysseyDesignTokens }) => ({
+    marginBlockStart: odysseyDesignTokens.Spacing2,
+  }),
+
+  ({ isDisabled, odysseyDesignTokens }) => ({
+    color: isDisabled ? odysseyDesignTokens.TypographyColorDisabled : "inherit",
+  }),
+
+  ({ isDisabled }) => ({
+    pointerEvents: isDisabled ? "none" : "auto",
+  }),
+);
```

Comment:
Curious, is there a reason these are all separate rather than one function return?

Reply:
I thought I read some where that putting them all in the same function return was less performant but, I don't have anything to back that up

Reply:
Ok cool, let's make it easier to read the code by using a single function return on these.